### PR TITLE
Improve making history (changelog)

### PR DESCRIPTION
**Pull request by slimreaper35**
## Change Summary
Changes apply to the script `release/make_history` which uses gitpython (GitPython) package. The file is no longer named `make_history.py` but `update_history.py`, which is optional.

## Related issue number
N/A

## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [x] Unit tests for the changes exist
* [x] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @davidhewitt

### CODE REVIEW

**Comment by slimreaper35** [comment]
please review

**Comment by slimreaper35** [comment]
please review

**Comment by slimreaper35** [comment]
please review

**Comment by samuelcolvin** [comment]
Definitely don't want to change the file name.

Personally I'm not that keen on changing unless there's a good reason.

More impactfully, @davidhewitt suggested some changes here - https://github.com/pydantic/pydantic/pull/7244#discussion_r1304585752, maybe we could use that approach instead?

File: pyproject.toml [hunk 1]
```diff
@@ -85,7 +85,8 @@ docs = [
     "black>=23.3.0",
     "pytest-examples>=0.0.10",
     "pydantic-settings>=2.0b1",
-    "pydantic-extra-types @ git+https://github.com/pydantic/pydantic-extra-types.git@main"
+    "pydantic-extra-types @ git+https://github.com/pydantic/pydantic-extra-types.git@main",
+    "gitpython",
```
Comment (samuelcolvin) [inline pyproject.toml:89]: This needs to be reflected in the lock file.

**Comment by adriangb** [comment]
@samuelcolvin I updated the lockfile, I'll let you do the final approval since you're more familiar with this part of the release process

**Comment by davidhewitt** [comment]
I'd prefer go with #7530 which makes an API request to Github to autogenerate the release notes.

**Comment by slimreaper35** [comment]
OK. You can close the PR.

**Comment by hramezani** [comment]
Thanks @slimreaper35 for your effort

**Pull request closed without merging**